### DISCUSSION TITLE
How should RPM "extensions" be handled?

### DISCUSSION THREAD

**Discussion by dralley** [rpm-rs/rpm#63] [General]
RPM has the concept of extension tags which do not exist within the header as entries in and of themselves, but rather they are constructed on the fly when requested if the user passed in the flags to allow them to be used.

https://github.com/rpm-software-management/rpm/blob/master/docs/manual/tags.md#extensions

These are basically just helpers which have been dropped into the same query system as the rest of the RPM tags.

Should we support these in a similar way, or should we split the functionality off into standalone functions?  On one hand the standalone function approach is more easily documented and more idiomatic to use.  On the other hand users who are porting code from C may have trouble understanding why they can fetch the values of certain tags from librpm but not from rpm-rs.

**Comment by drahnr** [comment]
The first question that pops to my mind with rpm features: Did you ever see the concept being used in practice?

**Reply by dralley** [reply]
They're definitely used in the wild.  Another good example is that certain "longsize" tags get polyfilled such that you can rely on them, even if the RPM itself only uses the original 32 bit size tags.

https://github.com/rpm-software-management/rpm/blob/master/docs/manual/large_files.md#64-bit-tags

IMO there are few enough of these that just using additional methods for them seems like it would be fine.

**Comment by dralley** [comment]
Filed an issue.
